dali/kernels/imgproc/colour_manipulation/brightness_contrast_test.cc

```diff
@@ -0,0 +1,213 @@
+cv::Mat_<T> to_mat(T *ptr, Roi roi, int rows, int cols) {
+  cv::Mat_<T> mat(rows, cols * nchannels, ptr);
+  cv::Rect rect(roi.x * nchannels, roi.y, roi.w * nchannels, roi.h);
+  auto roimat = cv::Mat_<T>(mat, rect);
+  roimat = roimat.clone();  // Make cv::Mat continuous
+  return roimat;
+}
+
+}  // namespace detail
+
+
+template<class InputOutputTypes>
+class BrightnessContrastTest : public ::testing::Test {
+ protected:
+  BrightnessContrastTest() {
+    input_.resize(dali::volume(shape_));
+  }
+
+
+  void SetUp() final {
+    std::mt19937_64 rng;
+    UniformRandomFill(input_, rng, 0., 10.);
+    calc_output();
+  }
+
+
+  std::vector<typename InputOutputTypes::in> input_;
+  std::vector<typename InputOutputTypes::out> ref_output_;
+  TensorShape<3> shape_ = {240, 320, 3};
+  typename InputOutputTypes::in brightness_ = 4;
+  typename InputOutputTypes::in contrast_ = 3;
+  static constexpr size_t ndims = 3;
+
+
+  void calc_output() {
+    for (auto in : input_) {
+      ref_output_.push_back(in * contrast_ + brightness_);
+    }
+  }
+};
+
+
+namespace test_types {
+
+template<class InputType, class OutputType>
+struct InputOutputTypes {
+  using in = InputType;
+  using out = OutputType;
+};
+
+using ArgTypes = std::tuple<uint8_t, int8_t, uint16_t, int16_t, int32_t, float>;
+using MyTypesTuple = detail::AllPairs<InputOutputTypes, ArgTypes, ArgTypes>;
+using GTestTypes = typename detail::TupleToGTest<MyTypesTuple>::type;
+
+}  // namespace test_types
+
+
+TYPED_TEST_SUITE(BrightnessContrastTest, test_types::GTestTypes);
+
+TYPED_TEST(BrightnessContrastTest, check_kernel) {
+  BrightnessContrast<kernels::ComputeCPU, typename TypeParam::in, typename TypeParam::out> kernel;
+  check_kernel<decltype(kernel)>();
+}
+
+
+TYPED_TEST(BrightnessContrastTest, SetupTestAndCheckKernel) {
+  BrightnessContrast<kernels::ComputeCPU, typename TypeParam::in, typename TypeParam::out> kernel;
+  KernelContext ctx;
+  InTensorCPU<typename TypeParam::in, this->ndims> in(this->input_.data(), this->shape_);
+  auto reqs = kernel.Setup(ctx, in, this->brightness_, this->contrast_);
+  auto sh = reqs.output_shapes[0][0];
+  ASSERT_EQ(this->shape_, sh);
+}
+
+
+TYPED_TEST(BrightnessContrastTest, RunTest) {
+  BrightnessContrast<kernels::ComputeCPU, typename TypeParam::in, typename TypeParam::out> kernel;
+  KernelContext ctx;
+  InTensorCPU<typename TypeParam::in, 3> in(this->input_.data(), this->shape_);
+  auto reqs = kernel.Setup(ctx, in, this->brightness_, this->contrast_);
+  auto out_shape = reqs.output_shapes[0][0];
+  vector<typename TypeParam::out> output;
+  output.resize(dali::volume(out_shape));
+  OutTensorCPU<typename TypeParam::out, this->ndims> out(
+          output.data(), out_shape.template to_static<this->ndims>());
+
+  kernel.Run(ctx, out, in, this->brightness_, this->contrast_);
+
+  ASSERT_EQ(this->ref_output_.size(), out.num_elements()) << "Number of elements doesn't match";
+  for (int i = 0; i < out.num_elements(); i++) {
```

Comment:
```suggestion
  Check(this->ref_output_[i], out);
```
This will check shapes, iterate all dimensions and limit number of errors printed.